### DISCUSSION TITLE
Advice on how to work with LLMs that generate content in the backend?

### DISCUSSION THREAD

**Discussion by mikealche** [liveblocks/liveblocks#1592] [Help]
Hey everybody! I love this product!

Currently, I'm building an editor that assists you with the help of an AI LLM (like 95% of other AI startups 😜)

My current stack is: 
* TipTap Editor (which uses y-prosemirror bindings), for the editor
* LiveBlocks for the synchronization of the Yjs doc.

However, I'm having a bit of trouble figuring out exactly how to make the LLM streaming feature work with Liveblocks.

Let me explain:
The LLM streams markdown content to my backend. 
From there I need to find a way to sync that content to the Yjs doc that's hosted in LiveBlocks.
However the conversion to a Yjs doc is not straightforward at all. Or at least I'm not figuring out the exact sequence of steps on how to convert a chunk of markdown (which may be invalid still, because remember it's just a chunk for now!) to the correct yXmlFragment, and from there to push it to liveblocks.

I think the only viable solution right now is to:
1. Have my users get the streamed markdown from the backend into their browser's client
2. Automatically populating the TipTap editor content with the `setContent` hook. Which will then do all the heavy lifting, transform it to a valid Yjs doc and push it to LiveBlocks (thanks to the Collaboration extension)

HOWEVER, this latter approach blocks the ability of generating content in the background. Meaning that if one of my users closes the editor's tab. The content stops being pushed to LiveBlocks.

Is there a correct or recommended way to stream content (markdown) from an LLM into an existing Yjs doc? 

Thanks!

**Comment by pro-vi** [comment]
Have you figured out a design for this type of problem? From my experience it's easier to do it on the client side but obviously it they close the Editor it stops working
